src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTimeOffset.cs

```diff
@@ -0,0 +1,278 @@
+                }
+            }
+            JsonWriterHelper.EscapeString(ref propertyName, ref span, firstEscapeIndexProp, out int written);
+            propertyName = span.Slice(0, written);
+
+            WriteStringByOptions(ref propertyName, value);
+
+            if (propertyArray != null)
+                ArrayPool<byte>.Shared.Return(propertyArray);
+        }
+
+        private void WriteStringByOptions(ref ReadOnlySpan<char> propertyName, DateTimeOffset value)
+        {
+            if (_writerOptions.Indented)
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                WriteStringIndented(ref propertyName, value);
+            }
+            else
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                WriteStringMinimized(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringByOptions(ref ReadOnlySpan<byte> propertyName, DateTimeOffset value)
+        {
+            if (_writerOptions.Indented)
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                WriteStringIndented(ref propertyName, value);
+            }
+            else
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                WriteStringMinimized(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringMinimized(ref ReadOnlySpan<char> escapedPropertyName, DateTimeOffset value)
+        {
+            int idx = WritePropertyNameMinimized(ref escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringMinimized(ref ReadOnlySpan<byte> escapedPropertyName, DateTimeOffset value)
+        {
+            int idx = WritePropertyNameMinimized(ref escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringIndented(ref ReadOnlySpan<char> escapedPropertyName, DateTimeOffset value)
+        {
+            int idx = WritePropertyNameIndented(ref escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringIndented(ref ReadOnlySpan<byte> escapedPropertyName, DateTimeOffset value)
+        {
+            int idx = WritePropertyNameIndented(ref escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringValue(DateTimeOffset value, ref int idx)
+        {
+            while (_buffer.Length <= idx)
```

Comment:
If the `while` is needed it seems like a maintenance problem, that if a new formatted type was being added that someone would easily write `if` here; so AdvanceAndGrow should have the `while`, if necessary.

(The only case where this while runs as second time is if AdvanceAndGrow sets _buffer to the empty buffer)